Report on changing Memberships

tmtmtmtm opened the pull request:
Add a section to the report on Memberships that have been added or removed. It's not 100% obvious yet exactly what format will be most useful for these, so for now try grouping by Term, and displaying only the person's name and start/end date (if applicable). Once we've seen it in action for a while it'll hopefully become apparent whether this is sufficient/suitable.

tmtmtmtm committed:
Report on changing Memberships

Add a section to the report on Memberships that have been added or
removed. It's not 100% obvious yet exactly what format will be most
useful for these, so for now try grouping by Term, and displaying only
the person's name and start/end date (if applicable). Once we've seen
it in action for a while it'll hopefully become apparent whether this
is sufficient/suitable.

tmtmtmtm reviewed (commented):
There are no tests here yet. I propose there's more value to be gained from getting this live straight away and then adding tests later. Reasoning:
1. We don't know yet what we actually want here, so I'm wary of coupling
   the tests too tightly to the current version
2. We can see it in action with live PRs via `PULL_REQUEST=17134 bundle exec
   rake pull_request_summary:print` - the main value of the test will be to
   prevent regressions.
3. I'd like to find a really good real live example from an actual PR to
   use as our test case (ideally 2 additions and one removal). A quick scan
   of historic ones didn't reveal such a case quickly, but putting this
   live now will let us spot such a case through the day-to-day updates.
4. This will instantly add value in reviewing, and would have revealed
   the problems with PRs like
   https://github.com/everypolitician/everypolitician-data/pull/17134 much
   earlier.

tmtmtmtm committed:
Require ruby >= 2.1

We now use `Array.to_h`, so need Ruby 2.1.0 onwards.

chrismytton commented on Gemfile:
```diff
@@ -1,4 +1,6 @@
 source 'https://rubygems.org'
 
+abort 'Ruby should be >= 2.1.0' unless RUBY_VERSION.to_f >= 2.1
```
I think we should also add this to the `.gemspec` file:

``` ruby
  spec.required_ruby_version = '>= 2.1.0'
```

Which uses [RubyGem's Ruby version support](http://guides.rubygems.org/specification-reference/#required_ruby_version=).

tmtmtmtm replied:
Also, or instead?

chrismytton commented on comment_template.md.erb:
```diff
@@ -78,6 +78,32 @@ No organizations added
+
+### Added
+<% if file.memberships_added.any? %>
+<% file.memberships_added.group_by(&:legislative_period_id).sort_by { |t, _| t }.each do |t, ms| %>
```
I think the `.sort_by { |t, _| t }` bit here can be simplified to `.sort`.

tmtmtmtm replied:
I'm finding the docs for `sort` on a Hash a little confusing. I think you're almost certainly right that this will DTRT, but I think it's worth being explicit here, more for people reading the code than for `ruby` itself :)

chrismytton commented on comment_template.md.erb:
```diff
@@ -78,6 +78,32 @@ No organizations added
+
+### Removed
+<% if file.memberships_removed.any? %>
+<% file.memberships_removed.group_by(&:legislative_period_id).sort_by { |t, _| t }.each do |t, ms| %>
```
As above, this can be simplified to `.sort`.

chrismytton reviewed (approved):
Couple of minor suggestions, but no blockers to merging 👍  🚀

tmtmtmtm committed:
Add a required_ruby_version to gem spec

We need this _as well as_ the version in the Gemfile, as Bundler
doesn't pay attention to this one, and will merrily let you `bundle
install` under Ruby 2.0.0.
